salt/master_utils/nodegroups.sls

```diff
@@ -0,0 +1,47 @@
+manage_nodegroups:
+  file.managed:
+    - name: /etc/salt/master.d/nodegroups.conf
+    - contents: |
+        nodegroups:
+            logging_cluster:
+              - 'G@roles:elasticsearch'
+              - 'and'
+              - 'G@environment:operations'
+            kibana:
+              - 'G@roles:kibana'
+              - 'and'
+              - 'G@environment:operations'
+            consul_operations:
+              - 'G@roles:consul_server'
+              - 'and'
+              - 'G@environment:operations'
+            consul_qa:
+              - 'G@roles:consul_server'
+              - 'and'
+              - 'P@environment:.*-qa$'
+            consul_production:
+              - 'G@roles:consul_server'
+              - 'and'
+              - 'P@environment:.*-production$'
```

Comment:
This wouldn't match the `production-apps` environment, and I'm not sure of the usefulness of targeting all consul servers in all production environments at once. Similarly for the rabbit nodegroup below. Typically we would address one environment's cluster at a time.

Reply:
I was attempting to define the `consul-servers-prod`, `consul-servers-qa`, and `rabbit-servers-prod` that are mentioned in https://github.com/mitodl/salt-ops/issues/859#issuecomment-533706185

I'm not that familiar with the consul and rabbitmq servers and how they'd typically be configured. Can you clarify what those groups should be; or should I disregard?

Reply:
I've just pushed some commits, based on our conversation today...